lib/db.js

```diff
@@ -349,35 +374,47 @@ module.exports = (
 
   DB.prototype.devices = function (uid) {
     log.trace({ op: 'DB.devices', uid: uid })
+    const promises = [
+      this.pool.get('/account/' + uid + '/devices')
+    ]
 
-    return this.pool.get('/account/' + uid + '/devices')
-      .then(
-        function (body) {
-          return body.map(function (item) {
-            return {
-              id: item.id,
-              sessionToken: item.sessionTokenId,
-              lastAccessTime: marshallLastAccessTime(item.lastAccessTime, uid, item.email),
-              name: item.name,
-              type: item.type,
-              pushCallback: item.callbackURL,
-              pushPublicKey: item.callbackPublicKey,
-              pushAuthKey: item.callbackAuthKey,
-              uaBrowser: item.uaBrowser,
-              uaBrowserVersion: item.uaBrowserVersion,
-              uaOS: item.uaOS,
-              uaOSVersion: item.uaOSVersion,
-              uaDeviceType: item.uaDeviceType
-            }
-          })
-        },
-        function (err) {
-          if (isNotFoundError(err)) {
-            throw error.unknownAccount()
+    if (this.redis) {
+      promises.push(this.redis.getAsync(uid))
+    }
+    return P.all(promises)
+      .then(results => {
+        log.info({op: 'devices hello', name: results})
+        const devices = results[0]
+        const tokens = results[1] ? JSON.parse(results[1]) : []
```

Comment:
`tokens` sounds strange, might be confusing for future readers. maybe `redisTokens`, `redisSession`, something like that?